### ISSUE TITLE
Google Drive API files.list request is limited to 8192 symbols

### ISSUE THREAD

**Issue by yetithefoot** [google/google-api-nodejs-client#972]
Not sure where to post, please reroute if not fits.

I'm working with huge queries to Google Drive `files.list` API endpoint. Handling millions of users' files I realized that HTTP GET requests URLs are limited to 8192 symbols. Sometimes I need to fetch files by thousands of ids or via huge search query. As a workaround I have to perform extra requests  that and stress Google Drive’s backend.

I tried to form instruct interpreter by passing POST+X-HTTP-Method-Override header. It works, but looks like limitation is somewhere on deeper level.

Is this 8192 symbols limit can be increased?

**Comment by JustinBeckwith** [comment]
👋 Greetings!  This is a way better answer than I could ever write:
https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers

TL;DR - browsers are generally limited to 2k characters in a url, and web servers generally stop responding after 8k characters.  These limits are unlikely to change.  

> As a workaround I have to perform extra requests that and stress Google Drive’s backend.

I don't think you need to worry about that :)  Your app would get rate limited before there ever was a problem there.   Sadly I don't think there's a lot we can do here, and certainly nothing node.js library  specific we could do.  I would suggest limiting the number of IDs you search for at a time, or using a different grouping criteria when making your query.
